src/icp_canister_backend/src/lib.rs

```diff
@@ -89,6 +90,11 @@ pub async fn deposit(user: Principal, timelock: u64) -> Result<(), types::Deposi
         return Err(types::DepositError::InsufficientBalance);
     };
 
+    // Check if deposit amount meets minimum requirement (must be enough for withdrawal)
+    if transfer_amount.0 < MINIMUM_DEPOSIT_AMOUNT.0 {
```

Comment:
Just replace in the line
if balance.0 > TRANSFER_FEE.0
TRANSFER_FEE with MINIMUM_DEPOSIT_AMOUNT